client/src/common/components/KohaiDetail.tsx

```diff
@@ -31,7 +32,14 @@ const KohaiDetail = () => {
 			? 'Contact is not provided yet'
 			: targetUser?.publicEmail;
 	};
-	return targetUser ? (
+
+	if (targetUser === null) {
+		return <LoadingScreen text="Loading" />;
+	}
+
+	return targetUser === undefined ? (
+		<Navigate to="/notfound" />
+	) : (
```

Comment:
I thought we would have to write some nested ternary operators, but this looks much simpler and is easy to understand what is going on here💡

Reply:
I couldn't do this without you and Taichi's help!

Maybe chaining ternary operators look smart, but I prefer this 😂